element_calcium_imaging/imaging_preprocess.py

```diff
@@ -0,0 +1,1283 @@
+    ):
+        param_dict = {
+            "processing_method": processing_method,
+            "paramset_idx": paramset_idx,
+            "paramset_desc": paramset_desc,
+            "params": params,
+            "param_set_hash": dict_to_uuid(params),
+        }
+        q_param = cls & {"param_set_hash": param_dict["param_set_hash"]}
+
+        if q_param:  # If the specified param-set already exists
+            pname = q_param.fetch1("paramset_idx")
+            if pname == paramset_idx:  # If the existed set has the same name: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    "The specified param-set already exists - name: {}".format(pname)
+                )
+        else:
+            cls.insert1(param_dict)
+
+
+@schema
+class CellCompartment(dj.Lookup):
+    definition = """  # Cell compartments that can be imaged
+    cell_compartment         : char(16)
+    """
+
+    contents = zip(["axon", "soma", "bouton"])
+
+
+@schema
+class MaskType(dj.Lookup):
+    definition = """ # Possible classifications for a segmented mask
+    mask_type        : varchar(16)
+    """
+
+    contents = zip(["soma", "axon", "dendrite", "neuropil", "artefact", "unknown"])
+
+
+# -------------- Trigger a processing routine --------------
+
+
+@schema
+class ProcessingTask(dj.Manual):
+    definition = """  # Manual table for defining a processing task ready to be run
+    -> PreProcess
+    -> ProcessingParamSet
+    ---
+    processing_output_dir: varchar(255)         #  output directory of the processed scan relative to root data directory
+    task_mode='load': enum('load', 'trigger')   # 'load': load computed analysis results, 'trigger': trigger computation
+    """
+
+    @classmethod
+    def infer_output_dir(cls, scan_key, relative=False, mkdir=False):
+        image_locators = {
+            "NIS": get_nd2_files,
+            "ScanImage": get_scan_image_files,
+            "Scanbox": get_scan_box_files,
+        }
+        image_locator = image_locators[(scan.Scan & scan_key).fetch1("acq_software")]
+
+        scan_dir = find_full_path(
+            get_imaging_root_data_dir(), image_locator(scan_key)[0]
+        ).parent
+        root_dir = find_root_directory(get_imaging_root_data_dir(), scan_dir)
+
+        paramset_key = ProcessingParamSet.fetch1()
+        processed_dir = pathlib.Path(get_processed_root_data_dir())
+        output_dir = (
+            processed_dir
+            / scan_dir.relative_to(root_dir)
+            / f'{paramset_key["processing_method"]}_{paramset_key["paramset_idx"]}'
+        )
+
+        if mkdir:
+            output_dir.mkdir(parents=True, exist_ok=True)
+
+        return output_dir.relative_to(processed_dir) if relative else output_dir
+
+    @classmethod
+    def auto_generate_entries(cls, scan_key, task_mode):
```

Comment:
Consider a shorter name, e.g. `generate`.  It has the same meaning as the longer name.

Reply:
See comment below.

Reply:
Thanks.  I have applied your suggestion with the following update to maintain backwards compatibility:
```
generate = auto_generate_entries
```